examples_extra/Placeholders/Sample/PostNode.m

```diff
@@ -60,6 +60,10 @@ - (instancetype)init
 - (UIImage *)placeholderImage
 {
   CGSize size = self.calculatedSize;
+  if (CGSizeEqualToSize(size, CGSizeZero)) {
+    return nil;
+  }
+  
   UIGraphicsBeginImageContext(size);
   [[UIColor colorWithWhite:0.9 alpha:1] setFill];
   UIRectFill((CGRect){CGPointZero, size});
```

Comment:
The first call is:
`-[ASDisplayView willMoveToWindow]` -->  `-[ASDisplayNode __enterHierarchy]` --> `-[ASDisplayNode _setupPlaceholderLayerIfNeeded]`

We setup the _placeholderLayer in this case but we don't assign a placeholderImage as the size is 0.

After layout is done we call `-[ASDisplayNode _completeLayoutCalculation]` and set the `_placeholderImage` in there. The problem is we don't assign it to the `_placeholderLayer` content:

```
if (_placeholderEnabled && [self _displaysAsynchronously] && self.contents == nil) {

  // Zero-sized nodes do not require a placeholder.
  CGSize layoutSize = (_layout ? _layout.size : CGSizeZero);
  if (CGSizeEqualToSize(layoutSize, CGSizeZero)) {
    return;
  }

  if (!_placeholderImage) {
    _placeholderImage = [self placeholderImage];
  }
}

```
The problem in there is we cannot set it in there as it has to be on the main thread.

The fix assign's a new right sized placeholder image after the layout is done that will be set to the already created _placeholderLayer's content